woocommerce/woocommerce-gutenberg-products-block
# Enable blockified templates

~~⚠️ This PR is blocked by #9579~~

This PR breaks E2E tests, so there are serial follow-up PRs to fix them:
- https://github.com/woocommerce/woocommerce-blocks/pull/9575
- https://github.com/woocommerce/woocommerce-blocks/pull/9577

Please, review them as well!

The goal of this PR is to render the new blocified templates, only when the user:
- installs the plugin.
- enables a block theme.

**and don't have a customized template.**

In this table, you can see the expected behavior for the different cases:


Plugin | Theme Type | Blockified block
-- | -- | --
New Installation | Classic Theme | 🚫
New Installation | Block | ✔️
New Installation | Switch from Classic Theme to Block Theme | ✔️
Update | Classic Theme | 🚫
Update | Block | 🚫
Update | Switch from Classic Theme to Block Theme | ✔️






Fixes #8209




### Testing

**it is not necessary to activate WC Blocks.**

#### Download zips:

1. Install the current version of WC Core from wp.org


2. Build of WooCommerce Core that contains a version of WC Blocks built with this PR. [woocommerce.zip](https://github.com/woocommerce/woocommerce-blocks/files/11574090/woocommerce.zip)



**Please, don't rename the name of the zip.**

For convenience, in the testing instructions:
- we use the name `pre-update package` for the plugin installed via wp.org.
- we use the name `after-update package` for the build of WooCommerce Core that contains a version of WC Blocks built with this PR.

#### User Facing Testing



### Test 1: 
Plugin | Theme Type | Blockified block
-- | -- | --
New Installation | Block | ✔️
1. Be sure that you have a block theme installed.
2. Install the plugin with the `after-update package`.
3. Open the FSE -> templates and check that  `Product Catalog`, `Product Search Results`, `Products by Category`. `Products by Tag` and `Single Product` are blockified.

### Test 2: 
Plugin | Theme Type | Blockified block
-- | -- | --
New Installation | Switch from Classic Theme to Block Theme  | ✔️
1. Be sure that you have a classic theme installed.
2. Install the plugin with the `after-update package`.
3. Active a block theme.
4. Open the FSE -> templates and check that  `Product Catalog`, `Product Search Results`, `Products by Category`, `Products by Tag` and `Single Product` are blockified.

### Test 3: 
Plugin | Theme Type | Blockified block
-- | -- | --
Update | Block | 🚫

1. Install `pre-update package`.
2. Open the FSE -> templates and check that  `Product Catalog`, `Product Search Results`, `Products by Category`, `Products by Tag` and `Single Product` renders the classic `WooCommerce Product Grid Block`. 
3. Update the plugin with `after-update package`.
4. Open the database, and check if in the table `wp_option` there is a key `wc_blocks_use_blockified_product_grid_block_as_template` set to `no`.
5. Open the FSE -> templates and check that  `Product Catalog`, `Product Search Results`, `Products by Category`, `Products by Tag` and `Single Product` renders the classic `WooCommerce Product Grid Block`.
6. Remove the key `wc_blocks_use_blockified_product_grid_block_as_template` and the key `wc_blocks_version` from the database and remove the plugin.

### Test 4: 
Plugin | Theme Type | Blockified block
-- | -- | --
Update | Switch from Classic Theme to Block Theme | ✔️

1. Be sure that you have a classic theme installed.
2. Install `pre-update package`.
3. Update the plugin with `after-update package`.
4. Enable a block theme.
5. Open the database, and check if in the table `wp_option` there is a key `wc_blocks_use_blockified_product_grid_block_as_template` set to `yes`.
6. Open the FSE -> templates and check that  `Product Catalog`, `Product Search Results`, `Products by Category`, `Products by Tag` and `Single Product` are blockified.
7. Remove the key `wc_blocks_use_blockified_product_grid_block_as_template` and the key `wc_blocks_version` from the database and remove the plugin.


### WooCommerce Visibility




### Performance Impact



### Changelog

> Enable blockified templates.
